src/main/java/org/wso2/extension/siddhi/window/uniquetimebatch/UniqueTimeBatchWindowProcessor.java

```diff
@@ -0,0 +1,379 @@
+import org.wso2.siddhi.core.util.collection.operator.MatchingMetaStateHolder;
+import org.wso2.siddhi.core.util.parser.OperatorParser;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+import org.wso2.siddhi.query.api.expression.Expression;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/*
+ * UniqueTimeBatch window
+ * Sample Query:
+ * from inputStream#window.unique:timeBatch(attribute,3 sec)
+ * select attribute1, attribute2
+ * insert into outputStream;
+ *
+ * Description:
+ * In the example query given, 3 is the duration of the window and attribute is the unique attribute.
+ * According to the given attribute it will give unique events within each given time batch.
+ * attribute true is to keep first unique default value is last unique.
+ *
+ * @since 1.0.0
+ */
+public class UniqueTimeBatchWindowProcessor extends WindowProcessor implements SchedulingProcessor,
+        FindableProcessor {
+
+    private long timeInMilliSeconds;
+    private long nextEmitTime = -1;
+    private ComplexEventChunk<StreamEvent> currentEventChunk = new ComplexEventChunk<>(false);
+    private ComplexEventChunk<StreamEvent> eventsToBeExpired = null;
+    private Map<Object, StreamEvent> uniqueEventMap = new HashMap<>();
+    private StreamEvent resetEvent = null;
+    private Scheduler scheduler;
+    private ExecutionPlanContext executionPlanContext;
+    private boolean isStartTimeEnabled = false;
+    private long startTime = 0;
+    private VariableExpressionExecutor uniqueKey;
+    private boolean isFirstUniqueEnabled = false;
+
+    /**
+     * The setScheduler method of the TimeWindowProcessor, As scheduler is private variable,
+     * to access publicly we use this setter method.
+     *
+     * @param scheduler the value of scheduler.
+     */
+    @Override
+    public void setScheduler(Scheduler scheduler) {
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * The getScheduler method of the TimeWindowProcessor, As scheduler is private variable,
+     * to access publicly we use this getter method.
+     */
+    @Override
+    public Scheduler getScheduler() {
+        return scheduler;
+    }
+
+    /**
+     * The init method of the WindowProcessor, this method will be called before other methods.
+     *
+     * @param attributeExpressionExecutors the executors of each function parameters
+     * @param executionPlanContext         the context of the execution plan
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors,
+                        ExecutionPlanContext executionPlanContext) {
+        this.executionPlanContext = executionPlanContext;
+        this.eventsToBeExpired = new ComplexEventChunk<>(false);
+        if (attributeExpressionExecutors.length == 2) {
+            uniqueKey = (VariableExpressionExecutor) attributeExpressionExecutors[0];
```

Comment:
As explained in https://github.com/wso2-extensions/siddhi-window-unique-lengthbatch/pull/1, let's put instanceof check here.